frame/contracts/src/lib.rs

```diff
@@ -671,6 +671,7 @@ decl_module! {
 			// If poking the contract has lead to eviction of the contract, give out the rewards.
 			if rent::try_evict::<T>(&dest, handicap) == rent::RentOutcome::Evicted {
 				T::Currency::deposit_into_existing(&rewarded, T::SurchargeReward::get())?;
+				Self::deposit_event(RawEvent::Evicted(dest));
```

Comment:
A contract can be evicted when it tries to pay its rent when it is called. I think this is missing, this should happen in exec.rs when it calls pay_rent.

Reply:
Okay thank you for telling me. I saw this and wasn't sure if that's actually the case.

Reply:
Should be implemented now. Please confirm correctness.